shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -113,6 +118,10 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool, val ne
     private def containsMth(key: (Str, Str) \/ (Opt[Str], Str)): Bool = mthEnv.contains(key) || parent.exists(_.containsMth(key))
     def containsMth(parent: Opt[Str], nme: Str): Bool = containsMth(R(parent, nme))
     def nest: Ctx = copy(Some(this), MutMap.empty, MutMap.empty)
+    // * Create a nested context for quasiquotes. If it is quoted, we need to clean `qenv` and `fvars`.
+    def qnest(quoted: Bool): Ctx =
+      if (quoted) copy(Some(this), MutMap.empty, MutMap.empty, inQuote = quoted, qenv = MutMap.empty, fvars = MutSet.empty)
+      else copy(Some(this), MutMap.empty, MutMap.empty, inQuote = quoted)
```

Comment:
Intuitively, shouldn't this branch just call `nest`? It seems wrong that within a quote, getting a nested _normal_ context (eg a lambda context) would switch `inQuote =  false`.

Reply:
The `qnest` is only for switching between quasiquotes and unquotes. I just documented it.

Reply:
That's a terrible API. Why not just have two functions called `enterQuote` and `enterUnquote` instead of a cryptic `qnest` contraption?